### PR TITLE
[IMP] workflows_engine Adds Wait task to workflows engine

### PR DESCRIPTION
**Pull request by BlommaertsEdwin**

### CODE REVIEW

File: src/core/tasks.py [hunk 1]
```diff
@@ -111,6 +112,65 @@ def as_dict(self):
...
+    def get_payload(self):
+        return self.payload
+
+    def get_payload_paths(self):
+        return self.payload_paths
+
+    def get_conditions(self):
+        return [c.identifier for c in self.conditions]
+
+    def get_validators(self):
```
Comment (nocturnalastro) [inline src/core/tasks.py]: you should have a super call here otherwise you will drop the preconditions

File: src/core/tasks.py [hunk 2]
```diff
@@ -225,7 +269,7 @@ class Condition(Task):
     __slots__ = [
         "conditions",
         "on_success",  # Flow step name on success
-        "success_message",
+        "success_message"Task
```
Comment (nocturnalastro) [inline src/core/tasks.py]: I don't think you meant to do this

File: src/core/tasks.py [hunk 3]
```diff
@@ -256,7 +300,7 @@ def get_message(message):
     @staticmethod
     def get_task_target(target):
         if target is None or target is False:
-            return target
+            return targetTask
```
Comment (nocturnalastro) [inline src/core/tasks.py]: same here

File: src/core/tests/tasks/test_wait.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+from ..schema_validator import get_validator_for
+from workflows_engine.core import tasks, components
+from workflows_engine import validators
+
+
+def test_jsonrpc():
```
Comment (nocturnalastro) [inline src/core/tests/tasks/test_wait.py]: don't forget to change the name

File: src/core/tasks.py [hunk 5]
```diff
@@ -157,6 +159,48 @@ def as_dict(self):
...
+            method=method, 
+            payload_paths=payload_paths, 
+            payload=payload,
+            response_path=response_path,
+            task_type="wait")
+        
+        self.timeout = timeout
+        self.conditions = conditions
+
+    def get_conditions(self):
```
Comment (nocturnalastro) [inline src/core/tasks.py:194]: you still need `get_validators` with a super call and this function in it.

**Review by nocturnalastro — CHANGES_REQUESTED** [review]

File: src/core/tasks.py (hunk 5 above)
Comment (BlommaertsEdwin) [reply src/core/tasks.py:194]: Yeh, didn't see it at first, you're right there :)

**Review by joe-gb — APPROVED** [review]

File: src/core/tests/tasks/test_wait.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        method="GET",
+        payload_paths=[
+            {"key": "$.arg1", "result_key": "$.arg1_result"},
+            {"key": "$.arg2", "result_key": "$.arg2_result"},
+        ],
+        payload={"arg1_result": None, "arg2_result": None},
+        response_path="$.response",
+    )
+    validator = get_validator_for("tasks/wait")
+    validator.validate(task.as_dict())
```
Comment (ws-mps) [inline src/core/tests/tasks/test_wait.py]: Missing new line end of file ?

File: src/core/tasks.py (hunk 5 above)
Comment (BlommaertsEdwin) [reply src/core/tasks.py:194]: @nocturnalastro Does the below function resolve this?

`    def get_validators(self):
        yield from super().get_validators()
        yield from self.conditions`

Comment (nocturnalastro) [reply src/core/tasks.py:194]: Yes that's what I meant sorry I didn't phrase it well.

btw try to use triple back ticks ``` when putting code in comments stops githubs formatting from f-ing it up

**Pull request merged**